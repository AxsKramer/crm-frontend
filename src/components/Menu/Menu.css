.nav-menu{
  display: flex;
  flex-direction: column;
  color: white;
}

.nav-menu a{
  padding: 0;
  margin: 2rem 0;
  font-size: 1.6rem;
}

.nav-menu a span{
  display: none;
}

.nav-menu a .icon{
  font-size: 2rem;
}

@media only screen and (min-width: 766px){
  .nav-menu a span{
    display: flex;
  }
  .nav-menu a{
    font-family: var(--fontText);
    display: grid;
    grid-template-columns: 30% 70%;
    justify-items: start;
    align-items: center;
  }
  .nav-menu a .icon{
    justify-self: center;
  }
}