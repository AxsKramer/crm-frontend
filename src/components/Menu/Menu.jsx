import React from "react";
import { NavLink } from "react-router-dom";
import './Menu.css'

const Menu = () => {
  return (
    <nav className="nav-menu">
      <NavLink
        to="/home"
        activeStyle={{
          fontWeight: "bold",
          color: "white",
        }}
      >
        <i className="fas fa-house-user icon"></i>
        <span title='Home'> Home</span>
      </NavLink>
      <NavLink
        to="users"
        activeStyle={{
          fontWeight: "bold",
          color: "white",
        }}
      >
        <i className="fas fa-user-plus icon"></i>
        <span title='New User'>New user</span>
      </NavLink>
      <NavLink
        to="/clients"
        activeStyle={{
          fontWeight: "bold",
          color: "white",
        }}
      >
        <i className="fas icon fa-users"></i>
        <span title='Clients'>Clients</span>
      </NavLink>
      <NavLink
        to="/products"
        activeStyle={{
          fontWeight: "bold",
          color: "white",
        }}
      >
        <i className="fas icon fa-boxes"></i>
        <span title='Products'>Products</span>
      </NavLink>
      <NavLink
        to="/orders"
        activeStyle={{
          fontWeight: "bold",
          color: "white",
        }}
      >
        <i className="far icon fa-copy"></i>
        <span title='Orders'>Orders</span>
      </NavLink>
      <NavLink
        to="/user"
        activeStyle={{
          fontWeight: "bold",
          color: "white",
        }}
      >
        <i className="fas icon fa-user-cog"></i>
        <span title='Settings'>Settings</span>
      </NavLink>
    </nav>
  );
};

export default Menu;
