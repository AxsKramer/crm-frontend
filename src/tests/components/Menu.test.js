import React from 'react';
import {shallow} from 'enzyme';
import Menu from '../../components/Menu/Menu';
// import '@testing-library/jest-dom';

describe("Testing of <Menu />", () => {
  const wrapper = shallow(<Menu />);

  test('should be displayed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('should have a nav tag ', () => {
    const nav = wrapper.find('nav').length;
    expect(nav).toBe(1);
  })

  test('nav tag should have class name "nav-menu" ', () => {
    const className = wrapper.find('nav').prop('className');
    expect(className).toBe('nav-menu');
  })

  test('should have NavLink instead of ancle link ', () => {
    const links = wrapper.find('NavLink').length;
    expect(links).toBeGreaterThanOrEqual(1);
  })

  test('should have the same icons as NavLinks', () => {
    const icons = wrapper.find('i').length;
    const NavLinks = wrapper.find('NavLink').length;
    expect(icons).toBeGreaterThanOrEqual(NavLinks);
  })

})