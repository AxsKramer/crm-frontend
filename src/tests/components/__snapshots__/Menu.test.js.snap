// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of <Menu /> should be displayed correctly 1`] = `
<nav
  className="nav-menu"
>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="/home"
  >
    <i
      className="fas fa-house-user icon"
    />
    <span
      title="Home"
    >
       Home
    </span>
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="users"
  >
    <i
      className="fas fa-user-plus icon"
    />
    <span
      title="New User"
    >
      New user
    </span>
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="/clients"
  >
    <i
      className="fas icon fa-users"
    />
    <span
      title="Clients"
    >
      Clients
    </span>
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="/products"
  >
    <i
      className="fas icon fa-boxes"
    />
    <span
      title="Products"
    >
      Products
    </span>
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="/orders"
  >
    <i
      className="far icon fa-copy"
    />
    <span
      title="Orders"
    >
      Orders
    </span>
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="/user"
  >
    <i
      className="fas icon fa-user-cog"
    />
    <span
      title="Settings"
    >
      Settings
    </span>
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "white",
        "fontWeight": "bold",
      }
    }
    to="/restore"
  >
    <i
      className="fas fa-trash-restore icon"
    />
    <span
      title="Settings"
    >
      Restore
    </span>
  </NavLink>
</nav>
`;
